toolchain/driver/compile_subcommand.cpp

```diff
@@ -521,20 +603,20 @@ auto CompilationUnit::RunParse() -> void {
   }
 }
 
-auto CompilationUnit::GetCheckUnit(SemIR::CheckIRId check_ir_id)
-    -> Check::Unit {
+auto CompilationUnit::GetCheckUnit() -> Check::Unit {
   CARBON_CHECK(parse_tree_, "Must call RunParse first");
   CARBON_CHECK(!sem_ir_, "Called GetCheckUnit twice");
 
   tree_and_subtrees_getter_ = [this]() -> const Parse::TreeAndSubtrees& {
     return this->GetParseTreeAndSubtrees();
   };
-  sem_ir_.emplace(&*parse_tree_, check_ir_id, parse_tree_->packaging_decl(),
-                  value_stores_, input_filename_);
+  sem_ir_.emplace(&*parse_tree_, SemIR::CheckIRId(unit_index_),
+                  parse_tree_->packaging_decl(), value_stores_,
+                  input_filename_);
   return {.consumer = consumer_,
           .value_stores = &value_stores_,
           .timings = timings_ ? &*timings_ : nullptr,
-          .tree_and_subtrees_getter = *tree_and_subtrees_getter_,
+          .check_ir_id = SemIR::CheckIRId(unit_index_),
```

Comment:
I was wondering about the comment on `unit_index_` being equivalent to CheckIRId, but then here we also convert from it twice. Why isn't the `unit_index_` a CheckIRId directly then?